File: src/models/Payment.js
// Payment model structure
class Payment {
  constructor(data = {}) {
    this.id = data.id || `payment_${Date.now()}`;
    this.appointmentId = data.appointmentId;
    this.patientId = data.patientId;
    this.doctorId = data.doctorId;
    this.amount = data.amount || 0;
    this.currency = data.currency || 'USD';
    this.status = data.status || 'PENDING'; // PENDING, COMPLETED, FAILED, REFUNDED
    this.paymentMethod = data.paymentMethod || 'CARD'; // CARD, PAYPAL, INSURANCE, etc.
    this.transactionId = data.transactionId;
    this.createdAt = data.createdAt || new Date().toISOString();
    this.updatedAt = data.updatedAt;
    this.cardDetails = data.cardDetails || {
      lastFourDigits: data.lastFourDigits || '',
      cardType: data.cardType || '',
      expiryMonth: data.expiryMonth || '',
      expiryYear: data.expiryYear || ''
    };
    // Insurance information if applicable
    this.insurance = data.insurance || {
      provider: data.insuranceProvider || '',
      policyNumber: data.insurancePolicyNumber || '',
      coveragePercentage: data.coveragePercentage || 0
    };
    // Additional metadata
    this.metadata = data.metadata || {};
  }
}

export default Payment; 

File: backend/src/main/java/com/hsa/model/Payment.java
package com.hsa.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "payments")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "appointment_id")
    private Long appointmentId;

    @Column(name = "patient_id")
    private Long patientId;

    @Column(name = "doctor_id")
    private String doctorId;

    @Column(nullable = false)
    private Double amount;

    @Column
    private String currency = "USD";

    @Column
    private String paymentMethod = "CASH"; // CASH, CARD, INSURANCE, etc.

    @Column
    private String status = "COMPLETED"; // PENDING, COMPLETED, REFUNDED

    @Column(name = "transaction_id")
    private String transactionId;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    public void prePersist() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
        if (this.currency == null) this.currency = "USD";
        if (this.status == null) this.status = "COMPLETED";
        if (this.paymentMethod == null) this.paymentMethod = "CASH";
    }

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}


